natss/pkg/dispatcher/dispatcher.go

```diff
@@ -135,17 +178,48 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 			}
 			return err
 		}
-		logger.Debugf("Published [%s] : '%s'", channel.String(), m.Headers)
+		logger.Debugf("Published [%s] : '%s'", channel.String(), event.String())
 		return nil
 	}
 }
 
-func (s *SubscriptionsSupervisor) Start(stopCh <-chan struct{}) error {
+func serialize(event *cloudevents.Event) ([]byte, error) {
```

Comment:
use the cloudevents sdk, build up the event and call `json.Marshal`

https://github.com/cloudevents/sdk-go

There are a lot of examples around the codebase.

Reply:
I updated the PR to use `event.MarshalJSON()` and `event.UnmarshalJSON(msg.Data)`.
Could you take another look?